### PR TITLE
appengine_flexible/pubsub: create topic if needed

### LINKED ISSUE #372: 09:03:22 CRITICAL: panic: runtime error: invalid memory address or nil pointer dereference
Doing the pub / sub tutorial for Go on Google App Engine Flexible environment listed here:

https://cloud.google.com/appengine/docs/flexible/go/writing-and-responding-to-pub-sub-messages

Follow everything as best as I can (which I believe is exact).  When I run it locally it will run, however when i try to create a message to send in the web interface it gives me the error above ( with other things).  Main code snippet below, though you can view total snippet at the above url.  Full Error below.  

```
func main() {
	ctx := context.Background()
	client, err := pubsub.NewClient(ctx, mustGetenv("GOOGLE_CLOUD_PROJECT"))
	if err != nil {
		log.Fatal(err)
	}

	// Create topic if it doesn't exist.
	topic, _ = client.CreateTopic(ctx, mustGetenv("PUBSUB_TOPIC"))

	http.HandleFunc("/", listHandler)
	http.HandleFunc("/pubsub/publish", publishHandler)
	http.HandleFunc("/pubsub/push", pushHandler)

	appengine.Main()
}
```

> 09:03:22 CRITICAL: panic: runtime error: invalid memory address or nil pointer dereference
>goroutine 26 [running]:
>google.golang.org/appengine/panic(0x149d8c0, 0x17ef8e0)
>	/usr/local/go/src/runtime/panic.go:491 +0x283
>cloud.google.com/go/pubsub.(*Topic).initBundler(0x0)
	/Users/ruben/go/src/cloud.google.com/go/pubsub/topic.go:327 +0x37
cloud.google.com/go/pubsub.(*Topic).Publish(0x0, 0x17ba580, 0xc4200160d0, 0xc4201d04d0, 0x0)
	/Users/ruben/go/src/cloud.google.com/go/pubsub/topic.go:248 +0x11d
main.publishHandler(0x17b8880, 0xc4201fccc0, 0xc4201a0200)
	/Users/ruben/go/src/github.com/GoogleCloudPlatform/golang-samples/appengine_flexible/pubsub/pubsub.go:101 +0x117
net/http.HandlerFunc.ServeHTTP(0x155d1c0, 0x17b8880, 0xc4201fccc0, 0xc4201a0200)
	/usr/local/go/src/net/http/server.go:1918 +0x44
net/http.(*ServeMux).ServeHTTP(0x17ffac0, 0x17b8880, 0xc4201fccc0, 0xc4201a0200)
	/usr/local/go/src/net/http/server.go:2254 +0x130
google.golang.org/appengine/internal.executeRequestSafely(0xc4201fccc0, 0xc4201a0200)
	/Users/ruben/go/src/google.golang.org/appengine/internal/api.go:153 +0x77
google.golang.org/appengine/internal.handleHTTP(0x17b9d00, 0xc4202dc1c0, 0xc4201a0200)
	/Users/ruben/go/src/google.golang.org/appengine/internal/api.go:121 +0x249
net/http.HandlerFunc.ServeHTTP(0x155cd40, 0x17b9d00, 0xc4202dc1c0, 0xc420258400)
	/usr/local/go/src/net/http/server.go:1918 +0x44
net/http.serverHandler.ServeHTTP(0xc4202904e0, 0x17b9d00, 0xc4202dc1c0, 0xc420258400)
	/usr/local/go/src/net/http/server.go:2619 +0xb4
net/http.(*conn).serve(0xc4200a1220, 0x17ba540, 0xc4201aa940)
	/usr/local/go/src/net/http/server.go:1801 +0x71d
created by ne
2018/01/18 09:03:22 Metadata fetch failed: Get http://metadata/computeMetadata/v1/instance/attributes/gae_project: dial tcp: lookup metadata: no such host
exit status 1

### PR DESCRIPTION
**Pull request by tbpg**
Fixes #372.

### CODE REVIEW

File: appengine_flexible/pubsub/pubsub.go [hunk 1]
```diff
@@ -39,10 +39,18 @@ func main() {
...
-	if topic == nil {
-		topic = client.Topic(topicName)
+	topic = client.Topic(topicName)
+	// Create the topic if it doesn't exist.
+	exists, err := topic.Exists(ctx)
+	if err != nil {
+		panic(err)
+	}
+	if !exists {
+		log.Println("t doesn't exist")
```
Comment (jadekler) [inline appengine_flexible/pubsub/pubsub.go]: How about, `log.Printf("Topic %v doesn't exist - creating it", topicName)`?

File: appengine_flexible/pubsub/pubsub.go [hunk 2]
```diff
@@ -39,10 +39,18 @@ func main() {
...
 	topicName := mustGetenv("PUBSUB_TOPIC")
-	topic, _ = client.CreateTopic(ctx, topicName)
-	// Get handle for topic in case it already existed.
-	if topic == nil {
-		topic = client.Topic(topicName)
+	topic = client.Topic(topicName)
+	// Create the topic if it doesn't exist.
+	exists, err := topic.Exists(ctx)
+	if err != nil {
+		panic(err)
```
Comment (jadekler) [inline appengine_flexible/pubsub/pubsub.go]: log.Fatal(err)

File: appengine_flexible/pubsub/pubsub.go [hunk 3]
```diff
@@ -39,10 +39,18 @@ func main() {
...
+	}
+	if !exists {
+		log.Println("t doesn't exist")
+		_, err = client.CreateTopic(ctx, topicName)
+		if err != nil {
+			panic(err)
```
Comment (jadekler) [inline appengine_flexible/pubsub/pubsub.go]: log.Fatal(err)

File: appengine_flexible/pubsub/pubsub.go (hunk 2 above)
Comment (tbpg) [reply appengine_flexible/pubsub/pubsub.go]: Done.

File: appengine_flexible/pubsub/pubsub.go (hunk 1 above)
Comment (tbpg) [reply appengine_flexible/pubsub/pubsub.go]: Done.

File: appengine_flexible/pubsub/pubsub.go (hunk 3 above)
Comment (tbpg) [reply appengine_flexible/pubsub/pubsub.go]: Done.

**Pull request merged**